src/adapters/LocalStorage.ts

```diff
@@ -2,13 +2,30 @@ import { SyncAdapter } from '../LowSync.js'
 
 export class LocalStorage<T> implements SyncAdapter<T> {
   #key: string
+  #localStorage: {
+    getItem: (key: string) => string | null
+    setItem: (key: string, data: string) => void
+  }
 
   constructor(key: string) {
     this.#key = key
+
+    this.#localStorage = {
+      getItem: (): null => null,
+      setItem: (): void => undefined,
+    }
+
+    if (typeof global !== 'undefined' && 'localStorage' in global) {
+      this.#localStorage = global.localStorage
+    }
+
+    if (typeof window !== 'undefined' && 'localStorage' in window) {
+      this.#localStorage = window.localStorage
+    }
   }
 
   read(): T | null {
-    const value = global.localStorage.getItem(this.#key)
+    const value = this.#localStorage.getItem(this.#key)
```

Comment:
it's better to define localStorage as a global in your ambient types. https://bobbyhadz.com/blog/typescript-declare-global-variable

probably also best to use globalThis, but I can't think of a context where that matters and you can use localStorage simultaneously.